### PR TITLE
resolves #17 use hexapdf to optimize PDF

### LINKED ISSUE #17: No cross reference links in generated pdf file
There are no cross reference links for included chapters in the generated pdf document. It's just text.

Also all items in TOC are not clickable.

Just to make it clear, cross reference links are working fine in the epub, however I'm not sure if it's an issue of the asciidoctor.

### PR DESCRIPTION
**Pull request by mojavelinux**
- gs now removes internal xrefs
- hexapdf generates smaller PDF objects, but doesn't optimize images

### CODE REVIEW

**Comment by mojavelinux** [comment]
I'm still playing with some of the options to see if we can get a smaller file for the screen media type.

**Comment by mojavelinux** [comment]
The problem with the file size is the cover. It's enormous, and it only gets bigger when embedded into the PDF. I optimized it to a JPG using the following command:

```
convert \
  -colorspace RGB \
  -interlace Plane \
  -sampling-factor 4:2:0 \
  -define jpeg:dct-method=float \
  -quality 75% \
  src/docs/asciidoc/images/cover.png \
  src/docs/asciidoc/images/cover.jpg
```

(see https://stackoverflow.com/questions/7261855/recommendation-for-compressing-jpg-files-with-imagemagick)

Now the screen PDF is < 1MB, which is the smallest it's ever been.

File: Gemfile [hunk 1]
```diff
@@ -2,4 +2,5 @@ source 'https://rubygems.org'
 
 gem 'coderay'
 gem 'Ascii85'
```
Comment (mojavelinux) [inline Gemfile]: Why is the Ascii85 gem included here?

**Comment by mraible** [comment]
At one point, I was getting an error if it wasn’t included.

**Pull request merged**